src/renderer/mod.rs

```diff
@@ -87,12 +87,24 @@ impl Renderer {
         }
     }
 
+    pub fn handle_scale_factor_update(&mut self, scale_factor: f64) {
+        self.shaper.update_scale_factor(scale_factor as f32);
+        self.update_font_dimensions();
+    }
+
     fn update_font(&mut self, guifont_setting: &str) {
-        if self.shaper.update_font(guifont_setting) {
-            let (font_width, font_height) = self.shaper.font_base_dimensions();
-            self.font_width = font_width;
-            self.font_height = font_height;
-        }
+        self.shaper.update_font(guifont_setting);
+        self.update_font_dimensions();
+    }
+
+    fn update_font_dimensions(&mut self) {
+        let (font_width, font_height) = self.shaper.font_base_dimensions();
+        self.font_width = font_width;
+        self.font_height = font_height;
+        println!(
```

Comment:
also here. Trace rather than println (note: i realize this is probably for testing, but is likely useful for debugging purposes, so marking these in the review)

Reply:
```suggestion
        trace!(
```